rebasehelper/build_helper.py

```diff
@@ -615,6 +616,154 @@ def build(cls, spec, sources, patches, results_dir, **kwargs):
                 'koji_task_id': koji_task_id}
 
 
+@register_build_tool
+class CoprBuildTool(BuildToolBase):
+    """
+    Class representing Copr build tool.
+    """
+
+    CMD = "copr"
+    logs = []
+    copr_helper = None
+
+    prefix = 'rebase-helper-'
+    chroot = 'fedora-rawhide-x86_64'
+    description = 'Repository containing rebase-helper builds.'
+    instructions = '''You can use this repository to test functionality
+                      of rebased packages.'''
+
+    @classmethod
+    def match(cls, cmd=None):
+        if cmd == cls.CMD:
+            return True
+        else:
+            return False
+
+    class CoprTemporaryEnvironment(BuildTemporaryEnvironment):
+        """
+        Class representing temporary environment for CoprBuildTool.
+        """
+
+        TEMPDIR_RPMBUILD = TemporaryEnvironment.TEMPDIR + '_RPMBUILD'
+        TEMPDIR_BUILD = TemporaryEnvironment.TEMPDIR + '_BUILD'
+        TEMPDIR_BUILDROOT = TemporaryEnvironment.TEMPDIR + '_BUILDROOT'
+        TEMPDIR_RPMS = TemporaryEnvironment.TEMPDIR + '_RPMS'
+        TEMPDIR_SRPMS = TemporaryEnvironment.TEMPDIR + '_SRPMS'
+
+        def _create_directory_sctructure(self):
+            # create rpmbuild directory structure
+            for dir_name in ['RESULTS', 'rpmbuild']:
+                self._env[self.TEMPDIR + '_' + dir_name.upper()] = os.path.join(
+                    self._env[self.TEMPDIR], dir_name)
+                logger.debug("Creating '%s'",
+                             self._env[self.TEMPDIR + '_' + dir_name.upper()])
+                os.makedirs(self._env[self.TEMPDIR + '_' + dir_name.upper()])
+            for dir_name in ['BUILD', 'BUILDROOT', 'RPMS', 'SOURCES', 'SPECS',
+                             'SRPMS']:
+                self._env[self.TEMPDIR + '_' + dir_name] = os.path.join(
+                    self._env[self.TEMPDIR_RPMBUILD],
+                    dir_name)
+                logger.debug("Creating '%s'",
+                             self._env[self.TEMPDIR + '_' + dir_name])
+                os.makedirs(self._env[self.TEMPDIR + '_' + dir_name])
+
+    @classmethod
+    def _build_srpm(cls, spec, workdir, results_dir):
```

Comment:
This seems to be the same code as in RpmbuildBuildTool class. It would be better to either inherint from the class and redefine the rest of the methods, or extract the SRPM building into a new class and then both classes should inherit from it. Duplicating of code is error prone.